Author: Nathan Phillip Brink <ohnobinki@ohnopublishing.net>
Purpose: Call $(CC) instead of $(LD) so that $(CFLAGS) can be
	included, fixing compilation for portage-multilib users. Also,
	add missing $(LDFLAGS), fixing a QA violation.

--- a/js/src/config/Linux_All.mk	Mon Jan 10 21:30:32 2011
+++ a/js/src/config/Linux_All.mk	Mon Jan 10 21:41:35 2011
@@ -44,8 +44,9 @@
 CFLAGS +=  -Wall -Wno-format
 OS_CFLAGS = -DXP_UNIX -DSVR4 -DSYSV -D_BSD_SOURCE -DPOSIX_SOURCE -DHAVE_LOCALTIME_R
 
+CCLD = $(CC)
 RANLIB = echo
-MKSHLIB = $(LD) -shared $(XMKSHLIBOPTS) -soname $(notdir $@)
+MKSHLIB = $(CCLD) $(CFLAGS) $(HOST_LDFLAGS) -shared $(XMKSHLIBOPTS) -Wl,-soname,'$(notdir $@)'
 
 #.c.o:
 #      $(CC) -c -MD $*.d $(CFLAGS) $<

Diff finished.  Mon Jan 10 21:41:36 2011
